packages/@react-spectrum/s2/src/Dialog.tsx

```diff
@@ -57,10 +57,13 @@ const header = style({
 
 const content =  style({
   flexGrow: 1,
+  flexShrink: {
+    [`@container (height < ${500 / 16}rem)`]: 0
+  },
   overflowY: {
     default: 'auto',
-    // Make the whole dialog scroll rather than only the content when the height it small.
-    [`@media (height < ${400 / 16}rem)`]: 'visible'
+    // Make the whole dialog scroll rather than only the content when the height is small.
+    [`@container (height < ${500 / 16}rem)`]: 'visible'
```

Comment:
Using container queries for this because when the keyboard opens the visible viewport resizes but the media query doesn't update.